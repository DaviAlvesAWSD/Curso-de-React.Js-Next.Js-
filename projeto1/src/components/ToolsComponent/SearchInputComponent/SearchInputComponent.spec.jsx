import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SearchInputComponent } from '.';

describe('<SearchInputComponent />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<SearchInputComponent handleChange={fn} searchValue={'testando'} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    expect(input.value).toBe('testando');
  });

  it('should call render handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<SearchInputComponent handleChange={fn} searchValue={'um valor qualquer'} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    const value = 'o valor';

    userEvent.type(input, value);
    expect(input.value).toBe('um valor qualquer');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<SearchInputComponent handleChange={fn} searchValue={''} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
