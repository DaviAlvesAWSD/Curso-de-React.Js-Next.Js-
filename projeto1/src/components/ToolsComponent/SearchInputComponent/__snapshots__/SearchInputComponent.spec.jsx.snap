// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInputComponent /> should match snapshot 1`] = `
<input
  class="search"
  placeholder="Type your search"
  type="search"
  value=""
/>
`;
